src/main/java/pl/net/rogala/eventy/service/EventService.java

```diff
@@ -2,23 +2,64 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Sort;
+import org.springframework.security.core.Authentication;
 import org.springframework.stereotype.Service;
 import pl.net.rogala.eventy.entity.Event;
+import pl.net.rogala.eventy.entity.User;
+import pl.net.rogala.eventy.form.NewEventForm;
 import pl.net.rogala.eventy.repository.EventRepository;
+import pl.net.rogala.eventy.repository.UserRepository;
 
+import java.time.LocalDate;
 import java.util.List;
 
 @Service
 public class EventService {
 
     private EventRepository eventRepository;
+    private UserRepository userRepository;
+
+    private UserService userService;
+
+    private NewEventForm eventForm;
 
     @Autowired
-    public EventService(EventRepository eventRepository) {
+    public EventService(EventRepository eventRepository, UserRepository userRepository, UserService userService, NewEventForm eventForm) {
         this.eventRepository = eventRepository;
+        this.userRepository = userRepository;
+        this.userService = userService;
+        this.eventForm = eventForm;
     }
 
     public List<Event> showEventList(){
         return eventRepository.findAll(Sort.by("startDate"));
     }
+
+    /**
+     * adding new event to database; setting logged user as owner of added event
+     * @param authentication gives logged user's e-mail
+     * @param eventForm form to adding new event
+     */
+    public void addNewEvent(NewEventForm eventForm, Authentication authentication){
+        Event event = new Event();
```

Comment:
Szczegóły implementacyjne takiego przepisywania przerzuciłbym do jednego z obiektów, tak aby móc wywoływać:
` Event event = eventForm.getEvent();`
Mocno poprawia to czytelność kodu.

Reply:
Tak, ale NewEventForm służy tylko do utworzenia formularza. Chyba, że stworzę nową, osobną klasę (poza serwisem), która będzie formularz przepisywać na event. Czyli chyba jakiś dodatkowy serwis, którego zadaniem będzie tylko przepisanie, a do którego później odwołam się w obecnym serwisie od eventów . Czy jest sens (albo może dobra praktyka), żeby to tak rozdzielić?

Reply:
Można to wydzielić do osobnej klasy i nazwać jako np. `EventConverter` którego jedyną rolą jest przepisywanie jednych obiektó na drugie. Jeżeli chcielibyście nieco zautomatyzować ten proces to można skorzystać z bibliotek do tego dedykowanych np [`Orika`](https://orika-mapper.github.io/orika-docs/) Radzę, uważać na zbyt zgłębokie przepisywania i zapętlenia.
Inny pomysł to sprawienie aby same obiekty były nieco bogatsze w logikę, poza samymi prostymi getterami i np. potrafiły tworzyć nowe obiekty jakiegoś typu na bazie danych któe posiadają.
W naszym konkretnym przypadku, dałoby się napisać czystą metodę, któa z obiektu formularza jest w stanie wytworzyć nowy obiekt modelowy lub encji z którym coś dalej będziemy robić.

Reply:
Stworzyłam nową klasę, która przepisuje formularz na event. Później w serwisie ten event jest już tylko zapisywany do bazy danych.